### DISCUSSION TITLE
How to properly set and initialize --user-data-dir with Camoufox and Playwright?

### DISCUSSION THREAD

**Discussion by xian-21** [daijro/camoufox#116] [Q&A]
I'm using Camoufox with Playwright in Python to automate browsing and need to specify a user data directory (--user-data-dir) for persistent sessions (e.g., login cookies). Here's a simplified code snippet.
**Problem**
The user_data_dir remains empty, and no files (e.g., Preferences, Cookies) are created.
One tab in the browser displays "Index of file:///..." instead of loading the intended URL.
**Question**
How can I ensure that Camoufox initializes and uses the user_data_dir correctly?
Why does the browser show the file index instead of navigating to the specified URL?

**Code**
`import asyncio
from camoufox import AsyncCamoufox

async def main():
    user_data_dir = r"C:\Users\user\OneDrive\桌面\python\threads\camoufox_user_data\shop_and_nextchat_browser_backup"

    async with AsyncCamoufox(headless=False, args=[f'--user-data-dir={user_data_dir}']) as browser:
        page = await browser.new_page()
        await page.goto("https://example.com")
        print("Page loaded")

if __name__ == "__main__":
    asyncio.run(main())
`

**Comment by 0x7357** [comment]
Firefox uses `-profile` instead of `--user-data-dir`.

**Comment by bjwktcsnzh** [comment]
@xian-21 For reference only:

https://github.com/Xpectuer/LibianCrawlers/blob/main/libiancrawlers/app_util/playwright_util.py#L89
